File: src/main/java/io/polygloat/dtos/request/EditSourceDTO.java
package io.polygloat.dtos.request;


import io.polygloat.dtos.PathDTO;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.validation.constraints.NotBlank;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class EditSourceDTO {
    @NotBlank
    private String oldFullPathString;

    @NotBlank
    private String newFullPathString;

    public PathDTO getOldPathDto() {
        return PathDTO.fromFullPath(oldFullPathString);
    }

    public PathDTO getNewPathDto() {
        return PathDTO.fromFullPath(newFullPathString);
    }
}

File: src/main/kotlin/io/polygloat/dtos/request/EditSourceDTO.kt
package io.polygloat.dtos.request

import io.polygloat.dtos.PathDTO
import javax.validation.constraints.NotBlank

data class EditSourceDTO(
        @field:NotBlank
        var oldFullPathString: String? = null,
        @field:NotBlank
        var newFullPathString: String? = null
) {

    val oldPathDto: PathDTO
        get() = PathDTO.fromFullPath(oldFullPathString)

    val newPathDto: PathDTO
        get() = PathDTO.fromFullPath(newFullPathString)

}